internal/controller/numaflowcontrollerrollout_controller.go

```diff
@@ -743,7 +688,22 @@ func (r *NumaflowControllerRolloutReconciler) processNumaflowControllerStatus(ct
 	if healthy {
 		controllerRollout.Status.MarkChildResourcesHealthy(controllerRollout.Generation)
 	} else {
-		controllerRollout.Status.MarkChildResourcesUnhealthy(conditionReason, conditionMsg, deployment.Generation)
+		controllerRollout.Status.MarkChildResourcesUnhealthy(conditionReason, conditionMsg, controllerRollout.Generation)
+	}
+
+	return nil
+}
+
+func (r *NumaflowControllerRolloutReconciler) markRolloutPaused(ctx context.Context, rolloutNamespace string, rolloutName string, paused bool) error {
+	rollout := &apiv1.NumaflowControllerRollout{}
+	if err := r.client.Get(ctx, k8stypes.NamespacedName{Namespace: rolloutNamespace, Name: rolloutName}, rollout); err != nil {
+		return err
+	}
+
+	if paused {
+		rollout.Status.MarkPausingPipelines(rollout.Generation)
+	} else {
+		rollout.Status.MarkUnpausingPipelines(rollout.Generation)
```

Comment:
found a bug (I'd written) here - was accidentally using `deployment.Generation` instead of `rollout.Generation`